File: Backend/src/main/java/lan/dk/podcastserver/config/JacksonConfig.java
package lan.dk.podcastserver.config;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.fasterxml.jackson.datatype.hibernate4.Hibernate4Module;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * Created by kevin on 15/06/2016 for Podcast Server
 */
@Configuration
public class JacksonConfig {

    @Bean
    public ObjectMapper mapper() {
        return new ObjectMapper()
                .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
                .registerModules(
                        new Hibernate4Module()
                            .enable(Hibernate4Module.Feature.FORCE_LAZY_LOADING)
                            .disable(Hibernate4Module.Feature.USE_TRANSIENT_ANNOTATION),
                        new JavaTimeModule()
                )
        ;
    }

}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/config/JacksonConfig.kt
package com.github.davinkevin.podcastserver.config

import org.springframework.boot.jackson.autoconfigure.JsonMapperBuilderCustomizer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import tools.jackson.databind.DeserializationFeature
import tools.jackson.databind.cfg.DateTimeFeature
import tools.jackson.databind.module.SimpleModule
import tools.jackson.databind.ser.std.ToStringSerializer
import tools.jackson.module.kotlin.kotlinModule
import java.nio.file.Path

/**
 * Created by kevin on 15/06/2016 for Podcast Server
 */
@Configuration
class JacksonConfig {

    @Bean
    fun mapperCustomization() = JsonMapperBuilderCustomizer {
        it
            .disable(DateTimeFeature.WRITE_DATES_AS_TIMESTAMPS)
            .addModules(
                SimpleModule("PathToString").apply { addSerializer(Path::class.java, ToStringSerializer.instance) },
                kotlinModule(),
            )
    }

}
